### DISCUSSION TITLE
2 barometers (one bmp280 internal fc matek F405 CTR, one external DPS310), certainly conflict i2c adress (0x76/0x77), how to solve simply ?

### DISCUSSION THREAD

**Discussion by esorven** [iNavFlight/inav#9667] [Q&A]
Hello,
I wanted to upgrade my quad (F405 CTR Matek) with a new gps (m10125+barometer+magnetometer) with better barometer (dsp310) on my quad.
On inav 7,  i took off the old  gps (+magnometer) and put the new gps (+magnometer+barometer dps310):
=> magnometer ok but barometer unavailable. If i disconnect the gps, i have baro ok (the bmp280 of the fc).
I think of a conflicting i2c adress. (the two barometer havec the same adress 0x76 or 0x77 ?)

How can i solve this simply ?
- I thought of flashing F405 OSD which have a emulated i2c bus (tx3 rx3) and put the gps (sda scl) on it =, will it work, same speed i2c ?
- Is there a simple method to change i2c adress of internal baro (shorting sdo to ground for example), i'm scared of short circuit ?
- How can i remove or cut a trace simply to disable internal baro (i will using external one) ?

If somebody have a solution please ?
Thanks.

**Comment by arturojs89** [comment]
Hi @esorven I have the same issue, did you figure out how to solve it?

**Comment by inquilinocdp** [comment]
You solved it? I'm facing the same questions...

**Reply by sensei-hacker** [reply]
If you're considering performing surgery on your flight controller, precisely how to do that depends quite a bit on your particular flight controller.

The general process would be:

1. Mask off all surrounding components 
2. Heat and remove the baro
3. Replace the nearby resistors and caps that accidentally moved
4. Wonder why it didn't make much difference 

Note that at low altitude, sonar, lidar, or fake "lidar" (actually just an LED) will all be more precise and accurate than the barometer.
